src/routes/eventstream.rs

```diff
@@ -1,39 +1,48 @@
 use axum::extract::State;
+use axum::http::HeaderMap;
 use axum::response::sse::{Event, Sse};
 use axum::routing::get;
 use axum::Router;
-use chrono::Utc;
-use futures::stream::Stream;
+use futures::stream::{self, Stream};
 use futures::StreamExt;
 use tokio_stream::wrappers::BroadcastStream;
 
 use crate::error::ApiResult;
 use crate::server::appstate::AppState;
 
 pub async fn get_clip_v2(
+    headers: HeaderMap,
     State(state): State<AppState>,
 ) -> Sse<impl Stream<Item = ApiResult<Event>>> {
     let hello = tokio_stream::iter([Ok(Event::default().comment("hi"))]);
+    let last_event_id = headers.get("last-event-id");
 
-    let mut prev_ts = Utc::now().timestamp();
-    let mut idx = 0;
-
-    let channel = state.res.lock().await.hue_channel();
+    let channel = state.res.lock().await.hue_event_stream().subscribe();
+    let stream = BroadcastStream::new(channel);
+    let events = match last_event_id {
+        Some(id) => {
+            let previous_events = state
+                .res
+                .lock()
+                .await
+                .hue_event_stream()
+                .events_sent_after_id(id.to_str().unwrap());
```

Comment:
Whoah, hey, no `.unwrap()` in the server code! :sweat_smile: 

This must be handled gracefully - I believe we already have error adaptation for `TryFromIntError`. Does `u64::try_from(id)?` work? (or whatever int type is appropriate)

Reply:
Oh shit, forgot to fix this properly :sweat_smile: 

Seems like using Rust to solve advent of code tasks (where crashing is most of the time the best thing to do) has made me too used to reading unwrap code :smile: